plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/webview/ToolkitLoginWebview.kt

```diff
@@ -188,6 +198,33 @@ class ToolkitWebviewBrowser(val project: Project) : LoginBrowser(project, Toolki
     }
 
     override fun prepareBrowser(state: BrowserState) {
+        selectionSettings.clear()
+
+        if (!inspectExistingConnection(project)) {
+            // existing connections
+            val bearerCreds = ToolkitAuthManager.getInstance().listConnections()
+                .filterIsInstance<AwsBearerTokenConnection>()
+                .associate {
+                    it.id to BearerConnectionSelectionSettings(it) { conn ->
+                        if (conn.isSono()) {
+                            loginBuilderId(CODECATALYST_SCOPES)
+                        } else {
+                            // TODO: rewrite scope logic, it's short term solution only
+                            val scopes = if (conn.isSono()) {
```

Comment:
Why do we check for isSono here? these are the Idc scopes right?

Reply:
oh yes this is redundant check